src/app/mail/mail-detail/mail-detail.component.ts

```diff
@@ -344,10 +346,16 @@ export class MailDetailComponent implements OnInit, OnDestroy {
                 this.scrollTo(document.querySelector('.last-child'));
               }
               // Mark mail as read
-              if (!this.mail.read && !this.markedAsRead) {
-                this.markedAsRead = false;
-                this.markAsRead(this.mail.id, false, false);
-              }
+              setTimeout(() => {
+                if (!this.mail.read && !this.markedAsRead) {
+                  this.markedAsRead = true;
+                  if (this.isAutoReadMode) {
+                    this.markAsRead(this.mail.id, true);
+                  } else {
+                    this.markAsRead(this.mail.id, false, false);
+                  }
+                }
+              }, 2000);
```

Comment:
@superdev0714 
How can you justify this timeout? I would expect instead you work with promises subscriptions, callback, or whatever triggers the method/function right after the event you expect to be processed within those 2000ms?

I can imagine random and difficult to debug issues because only some customers take longer than that to process.

Reply:
Please correct me if I'm wrong.